docs/architecture/blazor-for-web-forms-developers/migration.md

```diff
@@ -8,8 +8,322 @@ ms.date: 09/11/2019
+algorithms are considered unstable (where items that are equal may not be in the
+same order each time), but users may have taken a dependency on the ordering of
+the order of these unstable elements. In order to preserve this old behavior, a
+quirk was added that would continue to use the quick sort in later versions if
+the application targeted .NET Framework 4.0 but ran on .NET 4.5 or later.
+However, if the application were to be retargeted to .NET Framework 4.5 or
+later, then it would automatically opt into the introspective sort.
+
+The majority of these kinds of breaking changes are manifest when an application
+takes a dependency on an implementation detail (such as the undefined behavior
+of an unstable sort). To review what possible changes may impact your
+application, see the
+[documentation](https://docs.microsoft.com/dotnet/framework/migration-guide/application-compatibility)
+for this topic.
+
+### Enable Windows Compatibility Pack
+
+At this point, we are ready at a tooling level to make the switch to .NET
+Standard. However, there may still be APIs in the application that are not
+supported. Some of these types may be available via the [Windows Compatibility
+Pack](https://docs.microsoft.com/dotnet/core/porting/windows-compat-pack). This
+package adds support for around 20,000 APIs, including registry, WMI, Directory
+Services, among many more areas.
+
+Some of the high-level areas that are not supported in .NET Core:
+
+- AppDomains
+- Remoting
+- Code Access Security
+- Security Transparency
+- System.EnterpriseServices
+
+For more information on areas that are not supported, see the [documentation](https://docs.microsoft.com/dotnet/core/porting/net-framework-tech-unavailable).
+
+### Move to .NET Standard
+
+Having moved to SDK-style project files, moved to .NET 4.7.2 or later, and
+mitigated unsupported APIs, the project is now ready to be switched to .NET
+Standard. This change is as simple as changing:
+
+```
+<TargetFramework>net472</TargetFramework>
+```
+
+to 
+
+```
+<TargetFramework>netstandard2.0</TargetFramework>
+```
+
+If a .NET Framework version of the library is still needed for some reason (such
+as another dependency requiring it), the library can be multi-targeted so that
+both will be available when needed:
+
+```
+<TargetFrameworks>net472;netstandard2.0</TargetFrameworks>
+```
+
+Notice that here `TargetFrameworks` is used rather than `TargetFramework`. This
+is what is meant as multi-targeting and can be helpful while porting a large
+solution over to .NET Standard when some upstream components still need a .NET
+Framework build of the library.
+
+Once the target framework has been updated, the project has been updated to .NET
+Standard. This process will need to be continued for each project up the
+dependency tree as discovered by the .NET Portability Analyzer. Once all
+dependencies have been updated, then the application will be ready to be moved
+to Blazor.
+
+## Convert WebForms to Blazor
```

Comment:
I like that this section provides an overview of what it takes to migrate from Web Forms to Blazor, but I think it's missing some stuff.

Right now this section discusses:
- Async with no sync context
- Moving to new logging & config abstractions
- Startup

I think realistically we will initially need to recommend that folks start a migration by creating a new project. Ideally, we'd also have guidance on how to migrate an app in a piecemeal fashion (e.g. you migrate functionality a piece at a time). But given the complexities involved (two apps sharing the same URI space, auth cookies, etc.) I don't think it's something we can tackle at this point.

After creating the new project I think high level this is what migration looks like. I've bolded the steps that are Web Forms specific that I think we need to address. The rest as much as possible we should just mention and then point to docs.

- Update dependencies for .NET Core / .NET Standard
- Add the dependencies to the new project
- Migrate startup logic
- Migrate HTTP modules and handlers to middleware (point to doc)
- Migrate static files (should be a link)
- Migrate runtime bundling & minification setup (should be a link)
- **Migrate .aspx pages and add corresponding routes**
- **Implement web server control equivalents as needed or use components from the ecosystem**
- **Migrate user controls**
- **Migrate data access and binding**
- **Migrate forms with validation**
- Migrate config (should be a link)
- Leverage DI (should be a link)
- Address architectural differences (`HttpContext.Current`, no sync context, avoid blocking async class, no shadow copying, no request queue, single AppDomain in single process) (should be a link)

A big area that I think we're not really ready to address yet is authentication, identity, and authorization, so let's so that's out of scope right now.

Reply:
There are also some high level concerns to consider when moving from the server to the client. You may need to create endpoints for getting data, new security model, etc.